table/tables/partition.go

```diff
@@ -783,6 +816,38 @@ func (lp *ForListColumnPruning) LocatePartition(sc *stmtctx.StatementContext, v
 	return location, nil
 }
 
+// LocateRanges locates partition ranges by the column range
+func (lp *ForListColumnPruning) LocateRanges(sc *stmtctx.StatementContext, r *ranger.Range) ([]ListPartitionLocation, error) {
+	if r.LowVal[0].Kind() == types.KindMinNotNull {
+		r.LowVal[0] = types.GetMinValue(lp.ExprCol.GetType())
+	}
+	if r.HighVal[0].Kind() == types.KindMaxValue {
+		r.HighVal[0] = types.GetMaxValue(lp.ExprCol.GetType())
+	}
```

Comment:
Call this function modify the parameter `r` ?
This side effect is not good. How about:

```
if rowVal.Kind() == types.KindMinNotNull {
	lowVal = types.GetMinValue(lp.ExprCol.GetType())
}
if highVal.Kind() == types.KindMaxValue {
	highVal = types.GetMaxValue(lp.ExprCol.GetType())
}
```

Reply:
done